google/cloud/aiplatform/featurestore/featurestore.py

```diff
@@ -999,17 +984,204 @@ def batch_serve_to_gcs(
+            )
+
+        bigquery_client = bigquery.Client(
+            project=self.project, credentials=self.credentials
+        )
+
+        featurestore_name_components = self._parse_resource_name(self.resource_name)
+        featurestore_id = featurestore_name_components["featurestore"]
+
+        temp_bq_dataset_name = f"temp_{featurestore_id}_{uuid.uuid4()}".replace(
+            "-", "_"
+        )
+        temp_bq_dataset_id = f"{initializer.global_config.project}.{temp_bq_dataset_name}"[
+            :1024
+        ]
+        temp_bq_dataset = bigquery.Dataset(dataset_ref=temp_bq_dataset_id)
+        temp_bq_dataset.location = self.location
+        temp_bq_dataset = bigquery_client.create_dataset(temp_bq_dataset)
+
+        temp_bq_batch_serve_table_name = "batch_serve"
+        temp_bq_read_instances_table_name = "read_instances"
+        temp_bq_batch_serve_table_id = (
+            f"{temp_bq_dataset_id}.{temp_bq_batch_serve_table_name}"
+        )
+        temp_bq_read_instances_table_id = (
+            f"{temp_bq_dataset_id}.{temp_bq_read_instances_table_name}"
+        )
+
+        try:
+
+            job = bigquery_client.load_table_from_dataframe(
+                dataframe=read_instances_df, destination=temp_bq_read_instances_table_id
+            )
+            job.result()
+
+            self.batch_serve_to_bq(
+                bq_destination_output_uri=f"bq://{temp_bq_batch_serve_table_id}",
+                serving_feature_ids=serving_feature_ids,
+                read_instances_uri=f"bq://{temp_bq_read_instances_table_id}",
+                pass_through_fields=pass_through_fields,
+                feature_destination_fields=feature_destination_fields,
+                request_metadata=request_metadata,
+            )
+
+            bigquery_storage_read_client = bigquery_storage.BigQueryReadClient(
+                credentials=self.credentials
+            )
+            read_session_proto = bigquery_storage_read_client.create_read_session(
+                parent=f"projects/{self.project}",
+                read_session=bigquery_storage.types.ReadSession(
+                    table="projects/{project}/datasets/{dataset}/tables/{table}".format(
+                        project=self.project,
+                        dataset=temp_bq_dataset_name,
+                        table=temp_bq_batch_serve_table_name,
+                    ),
+                    data_format=bigquery_storage.types.DataFormat.ARROW,
+                ),
+            )
+
+        finally:
+            bigquery_client.delete_dataset(
+                dataset=temp_bq_dataset.dataset_id, delete_contents=True,
+            )
+
+        frames = []
+        for stream in read_session_proto.streams:
+            reader = bigquery_storage_read_client.read_rows(stream.name)
+            for message in reader.rows().pages:
+                frames.append(message.to_dataframe())
+
+        return None if not frames else pd.concat(frames)
```

Comment:
Should this use `ignore_index=True` or is the SOT for the index in the table or possibly `message.to_dataframe()` assigns incrementing index values?